components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMServiceTest.java

```diff
@@ -0,0 +1,618 @@
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+@PrepareForTest({DCRMService.class, ServiceProvider.class, IdentityProviderManager.class})
+public class DCRMServiceTest extends PowerMockTestCase {
+
+    private DCRMService dcrmService;
+
+    private OAuthAdminService mockOAuthAdminService;
+    private String dummyConsumerSecret = "dummyConsumerSecret";
+    private String dummyConsumerKey = "dummyConsumerKey";
+    private String dummyCallbackUrl = "dummyCallbackUrl";
+    private ApplicationRegistrationRequest mockApplicationRegistrationRequest;
+    private String dummyClientName = "dummyClientName";
+    private List<String> dummyGrantTypes = new ArrayList<>();
+    private String dummyUserName = "dummyUserName";
+    private String dummyTenantDomain = "dummyTenantDomain";
+    private ApplicationManagementService mockApplicationManagementService;
+    private ApplicationUpdateRequest mockApplicationUpdateRequest;
+    private String dummyClientId = "dummyClientId";
+
+    @BeforeTest
+    public void setUp() {
+
+        dcrmService = new DCRMService();
+    }
+
+    @DataProvider(name = "DTOProvider")
+    public Object[][] getDTOStatus() {
+
+        return new String[][]{
+                {null},
+                {""}
+        };
+    }
+
+    @Test
+    public void getApplicationEmptyClientIdTest() throws DCRMException {
+
+        try {
+            dcrmService.getApplication("");
+        } catch (IdentityException ex) {
+            assertEquals(ex.getMessage(), "Invalid client_id");
+            return;
+        }
+        fail("Expected exception IdentityException not thrown by getApplication method");
+    }
+
+    @Test(dataProvider = "DTOProvider")
+    public void getApplicationNullDTOTest(String dtoStatus) throws Exception {
+
+        mockOAuthAdminService = mock(OAuthAdminService.class);
+
+        if (dtoStatus == null) {
+            when(mockOAuthAdminService.getOAuthApplicationData(dummyConsumerKey)).thenReturn(null);
+        } else {
+            OAuthConsumerAppDTO dto = new OAuthConsumerAppDTO();
+            dto.setApplicationName("");
+            when(mockOAuthAdminService.getOAuthApplicationData(dummyConsumerKey)).thenReturn(dto);
+        }
+        Whitebox.setInternalState(dcrmService, "oAuthAdminService", mockOAuthAdminService);
+        try {
+            dcrmService.getApplication(dummyConsumerKey);
+        } catch (IdentityException ex) {
+            assertEquals(ex.getErrorCode(), DCRMConstants.ErrorMessages.NOT_FOUND_APPLICATION_WITH_ID.toString());
+            return;
+        }
+        fail("Expected exception IdentityException not thrown by getApplication method");
+    }
+
+    @Test
+    public void getApplicationDTOTestWithIOAException() throws Exception {
+
+        mockOAuthAdminService = mock(OAuthAdminService.class);
```

Comment:
It's better to use @Mock annotation from the beginning or move it to @BeforeTest method since this is used by many test methods.